### DISCUSSION TITLE
Heartbeat for Fetch doesn't seem to be working?

### DISCUSSION THREAD

**Discussion by tgilonis-hl** [nats-io/nats.go#1735] [Q&A]
Hi all, having some issues with the `jetstream.Consumer#Fetch` for `nats.go@v1.37.0` - our understanding is that if you specify a timeout of, say, 30 seconds, but a heartbeat period of 200 milliseconds, and you fail to receive 3 consecutive heartbeats at any point, the fetch call should return `ErrNoHeartbeat`. Is that correct? We are receiving no error under these conditions...

**Comment by Jarema** [comment]
@piotrpio will take a look, but please do not use such a short heartbeat values - it can put unnecessary strain on the server, and additionaly - it's really unreliable value in real-world scenarios with variadic RTT.

**Comment by piotrpio** [comment]
As @Jarema, this is not a realistic or suggested heartbeat value, but you should be receiving errors. Can you provide your usage or an example of when this is happening? We have tests for missing heartbeats using `Fetch` so I'm not sure in what scenario this may be happening.

As to how it works, missing heartbeats will be triggered if there are no messages and the client does not receive idle heartbeat message from the server for more than 2x specified value. This could happen e.g. if the consumer does not exist or there is a network issue. Are you sure the server is not sending the idle heartbeat messages?

**Reply by tgilonis-hl** [reply]
Thanks for the advice on heartbeat value, that definitely makes sense. We observed this happening when using a NATS cluster within K8S, scaling the NATS instance count down to 0, and then back up again. During that scaled down period we could not see any errors related to missing heartbeats.
